ct-app-unified/test/components/test_decorators.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+
+import pytest
+from core.components.baseclass import Base
+from core.components.decorators import connectguard, flagguard, formalin
+from core.components.lockedvar import LockedVar
+
+
+class FooClass(Base):
+    flag_prefix = "FOO_"
+
+    def __init__(self):
+        super().__init__()
+        self.connected = LockedVar("connected", False)
+        self.started = False
+
+    @connectguard
+    async def foo_connectguard_func(self):
+        await asyncio.sleep(0.5)
+
+    @flagguard
+    async def foo_flagguard_func(self):
+        await asyncio.sleep(0.5)
+
+    @flagguard
+    @formalin("Foo formalin")
+    async def foo_formalin_func(self):
+        await asyncio.sleep(0.5)
+
+
+@pytest.fixture
+def foo_class():
+    return FooClass()
+
+
+@pytest.mark.asyncio
+async def test_connectguard(foo_class: FooClass):
+    await foo_class.foo_connectguard_func()
+    pytest.skip("Not implemented")
```

Comment:
Is it going to be tested?

Reply:
Yep ! I added a bunch of method in `utils` that should be tested as well with some decorators.

Reply:
Decorator tested in [4973614](https://github.com/hoprnet/ct-research/pull/409/commits/49736149533c125ba8c48c18e065ecd0007b9aa7) (small fix in [a6ba661](https://github.com/hoprnet/ct-research/pull/409/commits/a6ba6616189ee9fcb5658023b1b1ffb88e9b514a)). Missing test on `Utils` methods will be added in a separate PR for simplicity. Code already running in staging using those methods